src/pages/on-chain-governance/index.page.tsx

```diff
@@ -1,74 +1,60 @@
-import React, { useEffect, useState } from "react";
+import React, { useState } from "react";
 import { isPlayground } from "@contexts/Environment";
-import { useRouter } from "next/router";
 import { Container } from "@components/commons/Container";
-import BigNumber from "bignumber.js";
-import {
-  CursorPage,
-  CursorPagination,
-} from "@components/commons/CursorPagination";
-import { NumericFormat } from "react-number-format";
-import {
-  getWhaleRpcClient,
-  newPlaygroundClient,
-  useWhaleRpcClient,
-} from "@contexts/WhaleContext";
+import { CursorPage } from "@components/commons/CursorPagination";
+import { getWhaleRpcClient, newPlaygroundClient } from "@contexts/WhaleContext";
 import { GetServerSidePropsContext } from "next";
 import {
   ListProposalsType,
   ListProposalsStatus,
-  ProposalType,
   ProposalInfo,
+  ProposalStatus,
 } from "@defichain/jellyfish-api-core/dist/category/governance";
 import { useNetwork } from "@contexts/NetworkContext";
 import { PlaygroundRpcClient } from "@defichain/playground-api-client";
+import { MdEdit, MdClose } from "react-icons/md";
+import classNames from "classnames";
 import { Link } from "@components/commons/link/Link";
-import { Button } from "./_components/Button";
-import { getDuration } from "./shared/durationHelper";
-import { ProgressBar } from "./_components/ProgressBar";
-import { votingStages } from "./enum/votingStages";
-import { OnChainGovernanceTitles } from "./enum/onChainGovernanceTitles";
-import { ProposalTable } from "./_components/ProposalTable";
+import { useWindowDimensions } from "hooks/useWindowDimensions";
+import { InfoHoverPopover } from "@components/commons/popover/InfoHoverPopover";
 import { ProposalCards } from "./_components/ProposalCard";
+import { ProposalTable } from "./_components/ProposalTable";
+import { Button } from "./_components/Button";
+import { getCurrentYearMonth } from "./shared/dateHelper";
 
 interface OCGProps {
-  votingCycle: {
-    votingCycleNumber: number;
+  allProposalsDetails: {
     proposalsSubmitted: number;
-    dfips: number;
-    cfps: number;
-    currentStage: votingStages;
-    timeLeft: number;
-    totalTime: number;
-    totalVotes: number;
+    openProposals: number;
+    closedProposals: number;
+    currentBlockCount: number;
+    currentBlockMedianTime: number;
+    userQueryProposalType: ListProposalsType;
+    userQueryProposalStatus: string | string[];
   };
   proposals: {
-    items: ProposalInfo[];
+    allProposals: ProposalInfo[];
+    queryProposals: ProposalInfo[];
     pages: CursorPage[];
   };
 }
-export default function OnChainGovernancePage(props) {
-  const rpc = useWhaleRpcClient();
-  const router = useRouter();
+export default function OnChainGovernancePage({
+  allProposalsDetails,
+  proposals,
+}) {
```

Comment:
if you do this, it's hard to catch error and missing the whole point of typescript, because `allProposalsDetails` and `proposals` are `type: any`.